code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java

```diff
@@ -40,14 +40,22 @@ public class MediaPublicTracker implements MediaTracker {
     private Map<String, Object> lastPlayheadParams;
 
     @VisibleForTesting
-    MediaPublicTracker(final String trackerId, final AdobeCallback<Event> eventConsumer) {
+    interface TimestampSupplier {
```

Comment:
nit: TimestampProvider

Reply:
I named the interface similar to Java's Functional Interfaces, https://docs.oracle.com/javase/8/docs/api/java/util/function/package-summary.html
Java provides a set of interfaces which may be used as targets for Lambda expressions. For example LongConsumer is an interface which takes a Long and returns no result, and LongSupplier is an interface which takes nothing and returns a Long. Unfortunately the minimum required Android API to use the functional interfaces is 24, which is why I created the TimestampSupplier.